src/map/ai/helpers/pathfind.cpp

```diff
@@ -152,23 +152,20 @@ bool CPathFind::PathInRange(const position_t& point, float range, uint8 pathFlag
 
     bool result = PathTo(point, pathFlags, false);
 
-    if (point.y - m_POwner->loc.p.y > 4) // Target is too low.
-    {
-        result = false;
-    }
-
-    if (m_POwner->objtype == TYPE_MOB) // Target is too high.
+    if (m_POwner->objtype == TYPE_MOB && !m_POwner->loc.zone->m_updatedNavmesh) // Target is too high.
     {
         auto PMob = static_cast<CMobEntity*>(m_POwner);
-        if (point.y - m_POwner->loc.p.y < -4)
+
+        if (point.y - m_POwner->loc.p.y < -6 && !PMob->PAI->IsRoaming())
         {
             auto disengageMod = PMob->getMobMod(MOBMOD_DISENGAGE_NO_PATH);
 
-            if (PMob->isInDynamis() && PMob->m_pathFindDisengage >= 1)
+            if (PMob->m_pathFindDisengage >= 1)
             {
                 result = false;
             }
-            else if (PMob->m_pathFindDisengage >= (disengageMod > 0 ? disengageMod : 2)) // This is just to stop players from abusing the disengage. Adjustable via mobmod.
+            else if ((PMob->m_pathFindDisengage >= (disengageMod > 0 ? disengageMod : 2)) ||
```

Comment:
Can we use enums here (and above) or is that just tedious?

Reply:
I mean there isn't much point to make enums since it is just a counter.